models/modelArticles.js

```diff
@@ -15,4 +15,37 @@ function retrieveArticleDataById(article_id) {
     });
 }
 
-module.exports = { retrieveArticleDataById };
+function retrieveAllArticleData(sort_by = "created_at", order = "DESC") {
+  const validOrders = ["ASC", "DESC"];
+
+  if (!validOrders.includes(order)) {
+    return Promise.reject({ status: 400, msg: "bad request" });
+  }
+
+  const validSortBy = ["author", "topic", "created_at"];
+
+  if (!validSortBy.includes(sort_by)) {
```

Comment:
You could probably factor this into the same `if` block as the validOrders one to keep your code dry